virtcontainers/factory/base/base.go

```diff
@@ -0,0 +1,20 @@
+// Copyright (c) 2018 HyperHQ Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+package base
+
+import vc "github.com/kata-containers/runtime/virtcontainers"
+
+// FactoryBase is vm factory's internal base factory interfaces.
+type FactoryBase interface {
+	// Config returns base factory config.
+	Config() vc.VMConfig
+
+	// GetBaseVM returns a paused VM created by the base factory.
+	GetBaseVM() (*vc.VM, error)
```

Comment:
The main advantage of a cache is of course speed. However, we need to be able to identify when the cache should be invalidated, for example when a new rootfs image containing a security fix is released.

As such, I wonder if we need to add a timestamp into `vc.VM` that shows when the cache item was created (not when  it was used). That will give us a basic way to look for "stale images" I think?

- https://github.com/kata-containers/runtime/blob/master/cli/list.go#L69
- https://github.com/clearcontainers/runtime/wiki/Stale-containers